### ISSUE TITLE
CDKTF: AWS WAFv2 rule statement changed to snake case, yet documentation remained in camel case

### ISSUE THREAD

**Issue by tianweiliu** [hashicorp/terraform-cdk#3637]
### Description

According to issue #3458, everything under `statement` block of `aws_wafv2_rule_group` or `aws_wafv2_web_acl` has been changed to snake case instead of camel case when using cdktf via typescript.

Although this changed was done on Jan 11, it was not reflected in any documentation. Even the newest version of the documentation still use camel case in both argument reference and examples. It seems to also breaks the hashtag based quick navigation within the page.

Argument Reference in current documentation (5.51.1):
>The statement` block supports the following arguments:
>
>[andStatement](https://registry.terraform.io/providers/hashicorp/aws/5.51.1/docs/resources/wafv2_web_acl?lang=typescript#andStatement) - (Optional) Logical rule statement used to combine other rule statements with AND logic. See [andStatement](https://registry.terraform.io/providers/hashicorp/aws/5.51.1/docs/resources/wafv2_web_acl?lang=typescript#and_statement-block) below for details.
...

Expected:
>The statement block supports the following arguments:
>
>[and_statement](https://registry.terraform.io/providers/hashicorp/aws/5.51.1/docs/resources/wafv2_web_acl?lang=typescript#and_statement) - (Optional) Logical rule statement used to combine other rule statements with AND logic. See [and_statement](https://registry.terraform.io/providers/hashicorp/aws/5.51.1/docs/resources/wafv2_web_acl?lang=typescript#and_statement-block) below for details.

Example in current documentation (5.51.1):
```
statement: {
  managedRuleGroupStatement: {
    name: "AWSManagedRulesCommonRuleSet",
    ruleActionOverride: [
      {
        actionToUse: {
          count: {},
        },
        name: "SizeRestrictions_QUERYSTRING",
      },
      {
        actionToUse: {
          count: {},
        },
        name: "NoUserAgent_HEADER",
      },
    ],
    scopeDownStatement: {
      geoMatchStatement: {
        countryCodes: ["US", "NL"],
      },
    },
    vendorName: "AWS",
  },
},
```

Expected:
```
statement: {
  managed_rule_group_statement: {
    name: "AWSManagedRulesCommonRuleSet",
    rule_action_override: [
      {
        action_to_use: {
          count: {},
        },
        name: "SizeRestrictions_QUERYSTRING",
      },
      {
        action_to_use: {
          count: {},
        },
        name: "NoUserAgent_HEADER",
      },
    ],
    scope_down_statement: {
      geo_match_statement: {
        country_codes: ["US", "NL"],
      },
    },
    vendor_name: "AWS",
  },
},
```

### Links

- https://registry.terraform.io/providers/hashicorp/aws/5.51.1/docs/resources/wafv2_web_acl?lang=typescript
- https://registry.terraform.io/providers/hashicorp/aws/5.51.1/docs/resources/wafv2_rule_group?lang=typescript

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by sabinayakc** [comment]
Yeah, figured out the hard way. It's a bit of a mess since one part of your code is camelCased and the other portion which are your statements are snake_cased.  For now I am using the snake_cased, and now will break in the future if anything changes.
